src/libsyncengine/jobs/network/API_v2/downloadjob.cpp

```diff
@@ -618,4 +528,171 @@ bool DownloadJob::moveTmpFile(bool &restartSync) {
+                                continue;
+                            } else {
+                                LOG_WARN(_logger, "Request " << jobId() << ": eof after reading " << getProgress()
+                                                             << " bytes from input stream");
+                                readError = true;
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (_vfsUpdateFetchStatus) {
+                    std::chrono::duration<double> elapsed_seconds = std::chrono::steady_clock::now() - fileProgressTimer;
+                    if (elapsed_seconds.count() > NOTIFICATION_DELAY || done) {
+                        // Update fetch status
+                        if (!_vfsUpdateFetchStatus(_tmpPath, _localpath, getProgress(), fetchCanceled, fetchFinished)) {
+                            LOGW_WARN(_logger, L"Error in vfsUpdateFetchStatus: " << Utility::formatSyncPath(_localpath));
+                            fetchError = true;
+                            break;
+                        } else if (fetchCanceled) {
+                            LOGW_WARN(_logger, L"Update fetch status canceled: " << Utility::formatSyncPath(_localpath));
+                            break;
+                        }
+                        fileProgressTimer = std::chrono::steady_clock::now();
+                    }
+                }
+            }
+        }
+    } else if (data) {
+        expectedSize = static_cast<std::streamsize>(data->get().length());
+        output.write(data->get().c_str(), expectedSize);
+    }
+
+    // Checks that the file has not been corrupted by another process
+    // Unfortunately, the file hash is not available, so we check only its size
+    output.flush();
+    output.seekp(0, std::ios_base::end);
+    if (expectedSize != Poco::Net::HTTPMessage::UNKNOWN_CONTENT_LENGTH && output.tellp() != expectedSize) {
+        LOG_WARN(_logger, "Request " << jobId() << ": tmp file has been corrupted by another process");
+        sentry::Handler::captureMessage(sentry::Level::Error, "DownloadJob::handleResponse", "Tmp file is corrupted");
+        writeError = true;
+    }
+
+    output.close();
+    if (output.bad()) {
+        // Read/writing error or logical error
+        LOG_WARN(_logger, "Request " << jobId() << ": error after closing tmp file");
+        writeError = true;
+    }
+
+    return true;
+}
+
+bool DownloadJob::createTmpFile(std::istream &is, bool &readError, bool &writeError, bool &fetchCanceled, bool &fetchFinished,
+                                bool &fetchError) {
+    return createTmpFile(std::make_optional<std::reference_wrapper<std::istream>>(is), std::nullopt, readError, writeError,
+                         fetchCanceled, fetchFinished, fetchError);
+}
+
+bool DownloadJob::createTmpFile(const std::string &data, bool &writeError) {
+    bool readError = false;
```

Comment:
```suggestion
    writeError = false;
    bool readError = false;
```
As a safety for the callers who would forget to initialize `writError`.

Reply:
Done into the base function:
```
bool DownloadJob::createTmpFile(std::optional<std::reference_wrapper<std::istream>> istr,
                                std::optional<std::reference_wrapper<const std::string>> data, bool &readError, bool &writeError,
                                bool &fetchCanceled, bool &fetchFinished, bool &fetchError) {
    assert(istr || data);

    readError = false;
    writeError = false;
    fetchCanceled = false;
    fetchFinished = false;
    fetchError = false;
```